source/extensions/filters/http/rbac/rbac_filter.h

```diff
@@ -40,12 +59,19 @@ class RoleBasedAccessControlFilterConfig {
 
   RoleBasedAccessControlFilterStats& stats() { return stats_; }
 
-  const Filters::Common::RBAC::RoleBasedAccessControlEngine&
-  engine(const Router::RouteConstSharedPtr route) const;
+  const absl::optional<Filters::Common::RBAC::RoleBasedAccessControlEngineImpl>&
+  engine(const Router::RouteConstSharedPtr route, EnforcementMode mode) const;
 
 private:
+  const absl::optional<Filters::Common::RBAC::RoleBasedAccessControlEngineImpl>&
```

Comment:
Can this derive from `RoleBasedAccessControlRouteSpecificFilterConfig`? all the code is basically the same.

Reply:
Since RoleBasedAccessControlRouteSpecificFilterConfig is derived from RouteSpecificFilterConfig , I may prefer to keep RoleBasedAccessControlFilterConfig as it is.